[NaychiMin] iP

NaychiMin opened the pull request.

NaychiMin committed:
Rename file from 'Duke.java' to 'Zran.java'

NaychiMin committed:
Add a new function 'echo'

The function echos inputs entered by the user and exits when the user inputs 'bye'.

NaychiMin committed:
Add new function 'addItems'

This function accepts user's input and adds it to a To Do list.

NaychiMin committed:
Add a new class named Task and modify existing functions to enable users to mark and unmark their tasks.

The class is used to manage each item the user inputs.

NaychiMin committed:
Add functions 'markAsDone()' and 'unmarkTask()' in Task.java

NaychiMin committed:
Add 'ToDo' class, 'Event' class and 'Deadline' class

janelleenqi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+        return (isDone? "X" : " ");
+    }
+
+    public void markAsDone(){
```
you can choose to combine this to become a setter method like "void setFound(boolean isFound);"

janelleenqi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
GOOD JOB for following the naming conventions (PascalCase for class and camelCase for variables)!!

janelleenqi commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+    public static void echo(String input, Task[] items) {
+        if(input.startsWith("mark ")) {
+            int itemIndex = Integer.parseInt(input.substring(5)) - 1;
+            System.out.printf((outputFormat) + "%n", "Congrats! :D \n    Task marked as done: " +
```
Wow! Good job keeping long lines short by using line wrapping!

janelleenqi commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+        addItems();
+    }
+
+    //The function 'echo' is used to print any output to the user
```
you did well by explicitly commenting on the purpose of functions you added!

bljhty commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+            System.out.printf((outputFormat) + "%n", "New task added: " + input + "\n" +
+                    "    Type 'list' to view all your tasks :D");
+        } else {
+            System.out.println("    ____________________________________________________________");
```
may be able to refactor the echo function to prevent code duplication

janelleenqi commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+                "   Type 'list' to view your to do list. \n " +
+                "   Type 'bye' to end my service. \n" +
+                "    Type 'mark' followed by item number to mark item as done. \n" +
+                "    Type 'unmark' followed by item number to unmark the item." );
```
i love your instructional introduction for easier usage by your users!

bljhty commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+
+    //The function 'echo' is used to print any output to the user
+    public static void echo(String input, Task[] items) {
+        if(input.startsWith("mark ")) {
```
may want to consider changing the name input to userInput

bljhty commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+                "   Type 'bye' to end my service. \n" +
+                "    Type 'mark' followed by item number to mark item as done. \n" +
+                "    Type 'unmark' followed by item number to unmark the item." );
+        addItems();
```
may want to replace the naming items to taskList

janelleenqi commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+                    "    Type 'list' to view all your tasks :D");
+        } else {
+            System.out.println("    ____________________________________________________________");
+            int index=0;
```
```suggestion
            int index = 0;
```

janelleenqi replied:
consider adding spaces to ensure consistency

bljhty commented on src/main/java/Zran.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.Scanner;
+
+public class Zran {
+    static String outputFormat = "    ____________________________________________________________\n" +
```
May like to rename this

janelleenqi reviewed (commented):
Good job! Your code was organised and very readable!

NaychiMin committed:
Add a new class 'TaskItems'.

'TaskItems' manages the lists of tasks including adding in of tasks and setting of task status.
Replace magic numbers with constants.

NaychiMin committed:
Add error handling using exceptions in TaskItems class and add a new class 'ZranExceptions' for custom exceptions.

NaychiMin committed:
Add ZranErrorMessages enum for custom error handling and to enhance error handling.

NaychiMin committed:
Add error handling for 'unmark' command.

NaychiMin committed:
Merge branch 'branch-Level-5'

woowenjun99 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+        this.isDone = false;
+    }
+
+    public String getDescription() {return (description);}
```
Perhaps you would like to review this single line method?

woowenjun99 commented on src/main/java/ZranErrorMessages.java:
```diff
@@ -0,0 +1,23 @@
+public enum ZranErrorMessages {
```
Nice to use enum!

woowenjun99 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,17 @@
+        super(description);
+        this.from = from;
+        this.to = to;
+        taskType = "E";
```
Perhaps you would like to consider using enum instead of magic string?

woowenjun99 commented on src/main/java/TaskItems.java:
```diff
@@ -0,0 +1,153 @@
+    public static final String USER_HELP_COMMAND = "help";
+
+    //to echo after a new task is added
+    public static void echo(Task task){
```
Perhaps you would like to relook at the spacing between the brackets?

woowenjun99 commented on src/main/java/TaskItems.java:
```diff
@@ -0,0 +1,153 @@
+    //to echo when LIST command is being used
+    public static void echo(Task[] items){
+        System.out.println("    ____________________________________________________________");
+        int index=0;
+        for(Task item : items){
```
Perhaps you would like to look at the spacing here?

woowenjun99 commented on src/main/java/TaskItems.java:
```diff
@@ -0,0 +1,153 @@
+
+        while(!(input = in.nextLine()).equals(EXIT_BOT_COMMAND)){
+            if(input.startsWith(MARK_TASK_COMMAND)) {
+                try{
+                    String eventIndex = input.substring(MARK_TASK_COMMAND.length()).trim();
+                    if(eventIndex.isEmpty()){
+                        throw new ZranExceptions(ZranErrorMessages.EMPTY_TASK_INDEX.message);
+                    }
+                    try{
+                        task = items[Integer.parseInt(eventIndex) - 1];
+                        task.setAsDone();
+                        echo(task, task.isDone);
+                    } catch(NullPointerException e) {
+                        throw new ZranExceptions(ZranErrorMessages.INVALID_TASK_INDEX.message);
+                    }
+                }
+                catch(ZranExceptions e){
+                    System.out.printf((outputFormat) + "%n", e.getMessage());
+                }
```
Perhaps you would like to use 2 catches instead of having one try catch block inside of another?

woowenjun99 commented on src/main/java/TaskItems.java:
```diff
@@ -0,0 +1,153 @@
+    }
+
+    // The function 'addTaskByType' is used to add an item into the task list by their task type
+    public static Task addTaskByType(String input) throws ZranExceptions{
```
Perhaps you would like to break this method into smaller methods as it is too long (i.e. more than 30 lines which is a violation of code quality)?

woowenjun99 reviewed (commented):
Good job on your first PR, perhaps you would like to consider breaking your methods down into smaller methods and replacing all the magic number and strings with variables and encapsulating the main logic into a single try catch block instead of multiple in one method

NaychiMin committed:
Refactor existing classes into three separate packages 'taskmanagement', 'app' and 'exceptions'.

- Move Deadline, Event, Task, TaskItems, ToDos to taskmanagement package.
- Move ZranExceptions and ZranErrorMessages to exceptions package.
- Move Zran to app package as the main application entry point.

NaychiMin committed:
Refactor code to use ArrayList instead of array in 'TaskItems.java'.

NaychiMin committed:
Add deleteTask function for task removal and error handling

Implemented the `deleteTask` function in the `TaskItems` class to remove a task from the task list.
The function also includes error handling for invalid task indices, such as NumberFormatException and IndexOutOfBoundsException.

NaychiMin committed:
Implement creating of text file to store data, load data from text file and save data to text file in a new class `TaskListData`.

NaychiMin committed:
Merge branch 'branch-Level-7'

NaychiMin committed:
Make the code more Object-Oriented Programming by introducing UI, Storage, Parser and TaskList classes for more modularity.

NaychiMin committed:
Increase code modularity for improved OOP structure

- Create UserInputValidator class to further process user's inputs and handle errors if any
- Reorganize functions in the Command package

NaychiMin committed:
Add LocalDate objects for users to key in deadlines and event dates in a specific date format ("yyyy-mm-dd").

- Create 'validateDate' function to further process user's inputs and handle errors if any.
- Ouput of LocalDate object is in the format "dd MMM yyyy".

NaychiMin committed:
Add 'FindTaskCommand' class to enable users to search for tasks with keywords.

- Experiment with the use of java Varargs feature

NaychiMin committed:
Merge pull request #1 from NaychiMin/branch-Level-8

Add date parsing for users to key in deadlines and event dates.

NaychiMin committed:
Merge branch 'master' into branch-Level-9

NaychiMin committed:
Merge pull request #2 from NaychiMin/branch-Level-9

Add function for users to filter tasks using keywords.

NaychiMin committed:
Add JavaDoc header comments for improved code documentation.

NaychiMin committed:
Merge branch 'master' into A-JavaDoc

NaychiMin committed:
Merge pull request #3 from NaychiMin/A-JavaDoc

Add JavaDoc header comments for improved code documentation.

NaychiMin committed:
Refactor addition of deadline and event tasks to support both standard and DateTime formatted inputs.

NaychiMin committed:
Update README.md

NaychiMin committed:
Update user guide with chatbot features and a summary of all recognised commands.

NaychiMin committed:
Update user guide and ammend typos.

NaychiMin committed:
Update user guide to be more informative.

NaychiMin committed:
Add quick start to user guide.

NaychiMin committed:
Edit formatting of User Guide.

NaychiMin committed:
Create _config.yml

NaychiMin committed:
Update README.md to fix minor typos

NaychiMin committed:
Edit command summary of README.md
